src/transactions/BumpSequenceTests.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright 2017 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "crypto/SignerKey.h"
+#include "lib/catch.hpp"
+#include "main/Application.h"
+#include "main/Config.h"
+#include "overlay/LoopbackPeer.h"
+#include "test/TestAccount.h"
+#include "test/TestExceptions.h"
+#include "test/TestUtils.h"
+#include "test/TxTests.h"
+#include "test/test.h"
+#include "transactions/TransactionFrame.h"
+#include "util/Logging.h"
+#include "util/Timer.h"
+#include "util/make_unique.h"
+
+using namespace stellar;
+using namespace stellar::txtest;
+
+TEST_CASE("bump sequence", "[tx][bumpsequence]")
+{
+    using xdr::operator==;
+
+    Config const& cfg = getTestConfig();
+
+    VirtualClock clock;
+    auto app = createTestApplication(clock, cfg);
+    app->start();
+
+    // set up world
+    auto root = TestAccount::createRoot(*app);
+    auto& lm = app->getLedgerManager();
+
+    auto a = root.create("A", lm.getMinBalance(0) + 1000);
+    auto b = root.create("B", lm.getMinBalance(0) + 1000);
+
+    SECTION("test success")
+    {
+        for_versions_from(10, *app, [&]() {
+            SECTION("small bump")
+            {
+                auto newSeq = a.loadSequenceNumber() + 2;
+                a.bumpSequence(newSeq);
+                REQUIRE(a.loadSequenceNumber() == newSeq);
+            }
+            SECTION("large bump")
+            {
+                auto newSeq = INT64_MAX;
+                a.bumpSequence(newSeq);
+                REQUIRE(a.loadSequenceNumber() == newSeq);
+                SECTION("no more tx when INT64_MAX is reached")
```

Comment:
so this is second way to lock account, next to setting master weight to 0, nice!

Reply:
no it doesn't lock the account, it just cannot be used as source account for a transaction